lib/s3.py

```diff
@@ -95,6 +100,26 @@ def sign_curl(url: urllib.parse.ParseResult, method='GET', checksum=SHA256_NIL,
     return [f'-H{key}:{value}' for key, value in headers.items()] + [url.geturl()]
 
 
+def urlopen(url: urllib.parse.ParseResult, method='GET') -> IO:
+    """Same as sign_request() but calls urlopen() on the result"""
+    headers = sign_request(url, method=method)
+    request = urllib.request.Request(url.geturl(), headers=headers, method=method)
+    return urllib.request.urlopen(request)
+
+
+def list_bucket(url: urllib.parse.ParseResult) -> ET:
```

Comment:
The code looks ok, but from the perspective of this commit it's dead code. I suggest adding a "list bucket contents" mode to main() here? That will both exercise the code, and moreover will be very useful for maintaining/inspecting the bucket.